pkg/registry/servicecatalog/broker/storage.go

```diff
@@ -58,7 +58,7 @@ func GetAttrs(obj runtime.Object) (labels.Set, fields.Set, error) {
 
 // NewStorage creates a new rest.Storage responsible for accessing Instance
 // resources
-func NewStorage(opts generic.RESTOptions) rest.Storage {
+func NewStorage(opts generic.RESTOptions) (brokers, brokersStatus rest.Storage) {
```

Comment:
That's exactly what the purpose is.  The status is not updateable via the normal endpoint to avoid updates to the resource destroying status information.  It's updated through the status subresource to allow differently permissioning end-users and service accounts that controllers run under.